web/packages/shared/components/ClusterDropdown/ClusterDropdown.tsx

```diff
@@ -0,0 +1,178 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { useHistory } from 'react-router';
+import { ButtonSecondary, Flex, Menu, MenuItem, Text } from 'design';
+import { ChevronDown } from 'design/Icon';
+import cfg from 'teleport/config';
+import { Cluster } from 'teleport/services/clusters';
+
+import { HoverTooltip } from 'shared/components/ToolTip';
+
+export interface ClusterDropdownProps {
+  clusterLoader: ClusterLoader;
+  clusterId: string;
+  /*
+   * onLoad is an optional prop. If onLoad is not passed, it will use the built in "loadClusters" function
+   */
+  onLoad?: () => Promise<Cluster[]>;
```

Comment:
i don't think we need the `onLoad` anymore right? i think we made it required to pass a fetcher

Reply:
I suppose we don't need it. I originally made it as a way to pass in _any_ func to load clusters but I guess I can't reason about when that would happen anymore. I'll remove it and it's references.